### DISCUSSION TITLE
CSO AVG is calculated for DATE_RANGE and not for DOY_RANGE

### DISCUSSION THREAD

**Discussion by dominiqueweber** [davidfrantz/force#198] [Q&A] [answered]
Hello

I was expecting CSO AVG to refer to the DOY_RANGE (like NUM) and not the DATE_RANGE. Is this the desired behavior or should I create an issue?

Update: Now I got it, the stats are calculated based on the MONTH_STEP parameter. But then my question is, would it not be nice to calculate avg temporal differences in CSO for a specific time range like 15th march to 15th august for 2019 (by ignoring MONTH_STEP for example if MONTH_STEP is 12)? Otherwise, as far as I understood, CSO number refers to the DATE_RANGE, but the other statistics are computed based on the MONTH_STEP. For me that was confusing.

Thanks, 
Dominique

**Comment by davidfrantz** [comment, marked as answer]
Hi Dominique,

if I am not mistaken, all the CSO parameters are calculated per MONTH_STEP - including NUM. Isn't this the case?

The only difference is: NUM is the pure number of available observations, and AVG (and all other) are referring to the time between consecutive observations.

In general, DATE_RANGE and DOY_RANGE are working together. With DATE_RANGE, you specify the general time frame. With DOY_RANGE, you can filter according to season (e.g. leave out winter images).

Cheers,
David

**Reply by dominiqueweber** [reply]
Hi David,

Thanks for your quick response. I guess you're right ;)

I was just confused because if you select a DOY_RANGE of let's say 100 - 200 with a MONTH_STEP of 12, then NUM is the pure number of CSO within that time, but, if I'm not mistaken, AVG is 365 (because of MONTH_STEP) divided by NUM.

Cheers,
Dominique
